solvcon/connection.py

```diff
@@ -193,6 +192,7 @@ def Client(address, family=None, authkey=None):
         try:
             skt.connect(address)
         except socket.error as e:
+            skt.close()
```

Comment:
This fixes the many `ResourceWarning`.

Reply:
A post-event note:

FYI, I used the following snippet to reduce the warnings:

```
import warnings

warnings.simplefilter("ignore", ResourceWarning)
```

Reply:
In my opinion, the current `skt.close()` should be good enough, and maybe better than `warnings.simplefilter("ignore", ResourceWarning)`? Because the latter closes the resources exactly. So I will suggest let us move forward.

Reply:
Yes, I think not closing the socket is the root cause of the warning and we should resolve it.